test/test_debugger.cpp

```diff
@@ -403,6 +419,23 @@ bool debugger_client::test_set_breakpoints()
     return rep["content"]["body"].size() != 0;
 }
 
+bool debugger_client::test_source()
+{
+    attach();
+
+    std::string code = make_code();
+    m_client.send_on_control("debug_request", make_dump_cell_request(4, code));
+    nl::json reply = m_client.receive_on_control();
+    std::string path = reply["content"]["body"]["sourcePath"].get<std::string>();
+
+    m_client.send_on_control("debug_request", make_source_request(5, path));
+    nl::json rep = m_client.receive_on_control();
+
+    nl::json source = rep["content"]["body"]["content"];
+    bool res = source == code + '\n';
```

Comment:
Do we need to add an extra `\n` in the `dumpCell` request?

https://github.com/QuantStack/xeus-python/blob/8faad74899564f267ef670f944f2285eb53e214f/src/xdebugger.cpp#L154